src/account/AsyncAccount.ts

```diff
@@ -0,0 +1,70 @@
+import type { AccountAuthenticator } from "../transactions/authenticator/account";
+import { HexInput, SigningScheme } from "../types";
+import type { AccountAddress } from "../core/accountAddress";
+import { AccountPublicKey, Signature, VerifySignatureArgs } from "../core/crypto";
+import { AnyRawTransaction } from "../transactions/types";
+
+/**
+ * Interface for a generic Aptos account.
+ *
+ * The interface is defined as abstract class to provide a single entrypoint for account generation,
+ * either through `Account.generate()` or `Account.fromDerivationPath`.
+ * Despite this being an abstract class, it should be treated as an interface and enforced using
+ * the `implements` keyword.
+ *
+ * Note: Generating an account instance does not create the account on-chain.
+ */
+export abstract class AsyncAccount {
+  /**
+   * Public key associated with the account
+   */
+  abstract readonly publicKey: AccountPublicKey;
+
+  /**
+   * Account address associated with the account
+   */
+  abstract readonly accountAddress: AccountAddress;
+
+  /**
+   * Signing scheme used to sign transactions
+   */
+  abstract signingScheme: SigningScheme;
+
+  /**
+   * Sign a message using the available signing capabilities.
+   * @param message the signing message, as binary input
+   * @return the AccountAuthenticator containing the signature, together with the account's public key
+   */
+  abstract signWithAuthenticator(message: HexInput): AccountAuthenticator | Promise<AccountAuthenticator>;
```

Comment:
I would just return a Promise only.  That way you always know you have to await on it (even if it doesn't make any special call).

Otherwise you have to do a lot of "if promise" await, "else do nothing"

Reply:
why? if a dev wants to use the `Account` type they should use the Account class then. This can be annoying for external users when more of them start using the tmp branch